test/object-store/migrations.cpp

```diff
@@ -913,817 +833,218 @@ TEST_CASE("migration: Automatic") {
-            REQUIRE(child_table->is_embedded());
-            REQUIRE(child_table->size() == 0);
-            REQUIRE(parent_table->size() == 0);
+            realm->begin_transaction();
+            parent_obj.set_any("mixed", Mixed());
+            parent_obj.get_list<Mixed>("list").clear();
+            parent_obj.get_set<Mixed>("set").clear();
+            parent_obj.get_dictionary("dictionary").clear();
+            realm->commit_transaction();
+
+            REQUIRE_UPDATE_SUCCEEDS(*realm, embedded_schema, 2);
         }
-        SECTION("change table from top-level to embedded, migration allowed, embedded object with 1 incoming link. "
-                "Resolve automatic "
-                "should not be triggered") {
-            InMemoryTestFile config;
-            config.automatic_handle_backlicks_in_migrations = true;
-            Schema schema = {
-                {"child_table",
-                 {
-                     {"value", PropertyType::Int},
-                 }},
-                {"parent_table",
-                 {
-                     {"child_property", PropertyType::Object | PropertyType::Nullable, "child_table"},
-                 }},
-            };
+
+        SECTION("automatic migration to embedded deletes objects with no incoming links") {
+            config.automatically_handle_backlinks_in_migrations = true;
+            config.schema = basic_link_schema;
             auto realm = Realm::get_shared_realm(config);
-            realm->update_schema(schema, 1);
-            auto child_table = ObjectStore::table_for_object_type(realm->read_group(), "child_table");
-            auto parent_table = ObjectStore::table_for_object_type(realm->read_group(), "parent_table");
+
+            auto child_table = ObjectStore::table_for_object_type(realm->read_group(), "child");
             realm->begin_transaction();
-            Obj child_object = child_table->create_object();
-            child_object.set("value", 42);
-            auto child_object_key = child_object.get_key();
-            parent_table->create_object().set_all(child_object_key);
+            child_table->create_object();
             realm->commit_transaction();
-            REQUIRE_FALSE(child_table->is_embedded());
-            REQUIRE(child_table->size() == 1);
-            REQUIRE(parent_table->size() == 1);
 
-            REQUIRE_NOTHROW(realm->update_schema(set_table_type(schema, "child_table", ObjectType::Embedded), 2,
-                                                 [](auto, auto, auto&) {}));
+            REQUIRE_UPDATE_SUCCEEDS(*realm, basic_embedded_schema, 1);
+            REQUIRE(child_table->size() == 0);
+        }
 
-            REQUIRE(realm->schema_version() == 2);
-            REQUIRE(child_table->is_embedded());
-            REQUIRE(child_table->size() == 1);
-            REQUIRE(parent_table->size() == 1);
-        }
-        SECTION("change table to embedded - multiple incoming links - resolved automatically + copy array of mixed "
```

Comment:
Most of these deleted tests are now tested in core, and only the tests which are actually testing something about the object store migration logic remain here.